src/core/thread/mle_router.cpp

```diff
@@ -50,7 +50,9 @@ namespace Mle {
 
 MleRouter::MleRouter(ThreadNetif &aThreadNetif):
     Mle(aThreadNetif),
-    mAdvertiseTimer(aThreadNetif.GetIp6().mTimerScheduler, &HandleAdvertiseTimer, this),
+    mRouterAdvertiseTimer(aThreadNetif.GetIp6().mTimerScheduler, TrickleTimer::kModeNormal,
+                          HandleRouterAdvertiseTimer, NULL, this),
+    mReedAdvertiseTimer(aThreadNetif.GetIp6().mTimerScheduler, &HandleReedAdvertiseTimer, this),
```

Comment:
I tried doing it, but now a test is failing. I think it comes down to the fact that the TrickleTimer class is actually waiting not waiting [`kReedAdvertiseInterval` to `kReedAdvertiseInterval + kReedAdvertiseJitter`]. It does set `I` equal to that, but it then sets `t` equal to [`I/2` to `I`].

The best way to fix that would be to add a new `Mode` for the REED timer, and instead of passing mode in the constructor, pass it in `Start`. I will try it out to see if it fixes the tests.